### ISSUE TITLE
Proper code for ddp training in Lightly 

### ISSUE THREAD

**Issue by AricYang** [lightly-ai/lightly#1458]
Hi, I'm training my SSL model in muti GPU settings. I wonder what is the proper way to set up the multi GPU training?
following is my code.

# defining model structure:
```
seed_everything(42, workers=True)
class TiCo_MP(pl.LightningModule):
    def __init__(self):
        super().__init__()
        resnet = torchvision.models.resnet18()
        self.backbone = nn.Sequential(*list(resnet.children())[:-2])
        self.projection_head = TiCoProjectionHead(512, 512, 128)
        self.criterion = TiCoLoss(gather_distributed=True,
        )                         
    def forward(self, x):
        ...
        return x
    def training_step(self, batch, batch_index):
        ...
        return loss
    def configure_optimizers(self):
        optimizer = ...
```
        
# defining trainer:
```
trainer = pl.Trainer(max_epochs=500,
                     accelerator = 'gpu',
                     devices = [0,1,2],
                     strategy="ddp_find_unused_parameters_false",
                     default_root_dir='...',
                     sync_batchnorm=True,
                     deterministic=True,
                     accumulate_grad_batches=32)
```                     
                     
 I accidentally train SSL without gather_distributed=True statement, end up having better performance for my SSL encoder with the training time of using 3 GPUs.
 My precise question is, is stating gather_distributed=True when defining the loss function necessary in my case? or defining it in the trainer  is enough?

**Comment by guarin** [comment]
Closing for now, please let me know if you have further questions :)
